nestkernel/simulation_manager.cpp

```diff
@@ -443,20 +507,39 @@ nest::SimulationManager::simulate( Time const& t )
       "is called repeatedly with simulation times that are not multiples of "
       "the minimal delay." );
 
-  resume_( num_active_nodes );
+  start_updating_();
+}
 
-  finalize_simulation_();
+void
+nest::SimulationManager::cleanup()
+{
```

Comment:
What do you think about a state flag within SimulationManager, which enables the execution of run()? It could be set to true within prepare() and afterwards set to false within cleanup().